File: my-folder/0169-majority-element/solution.ts
function majorityElement(nums: number[]): number {

    if (nums.length === 1) return nums[0];

    const map = new Map();
    for (let i of nums) {
        if (map.has(i)) {
            if (map.get(i) + 1 > Math.floor(nums.length / 2)) return i;
            map.set(i, map.get(i) + 1);
        } else {
            map.set(i, 1);
        }
    }
};

File: my-folder/0169-majority-element/solution.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        majority_elem = frequency = 0

        for n in nums:
            if frequency == 0:
                majority_elem = n
                frequency += 1
            elif majority_elem == n:
                frequency += 1
            else:
                frequency -= 1
        
        return majority_elem
                
